bindings/python/setup.py

```diff
@@ -187,24 +203,38 @@ def build_extension_with_b2(self):
         if self.cxxstd:
             args.append(f"cxxstd={self.cxxstd}")
 
-        # Jamfile hack to copy the module to our target directory
+        # Jamfile hacks to ensure we select the python environment defined in
+        # our project-config.jam
+        args.append("libtorrent-python=on")
+        args.append(f"python={sysconfig.get_python_version()}")
+
+        # Our goal is to produce an artifact at this path. If we do this, the
+        # distutils build system will skip trying to build it.
         target = pathlib.Path(self.get_ext_fullpath("libtorrent"))
+        self.announce(f"target: {target}")
+
+        # b2 doesn't provide a way to signal the name or paths of its outputs.
+        # We try to convince python.jam to name its output file like our target
+        # and copy it to our target directory.
+
+        # Jamfile hack to copy the module to our target directory
         args.append(f"python-install-path={target.parent}")
-        args.append("libtorrent-python=on")
         args.append("install_module")
 
         # We use a "project-config.jam" to instantiate a python environment
         # to exactly match the running one.
         config = tempfile.NamedTemporaryFile(mode="w+", delete=False)
         try:
-            write_b2_python_config(target, config)
+            write_b2_python_config(config)
             config.seek(0)
             self.announce("project-config.jam contents:")
             self.announce(config.read())
-            args.append(f"python={sysconfig.get_python_version()}")
+            config.close()
```

Comment:
do you need to close the file here too? It definitely needs to be closed in the finally block, in case an exception is thrown

Reply:
Doesn't Windows barf reading a file if another process has it open for writing? I didn't test with this removed.

double-closing is a no-op in python, so I thought this was a good defense.

Reply:
when I tested this on windows 10, I did not need to close it here, before invoking `b2`

Reply:
I suppose the pythonic way of doing this would be to introduce a scope:

```
with tempfile.NamedTemporaryFile(mode="w+", delete=False) as config:
   ...
```

Reply:
> when I tested this on windows 10, I did not need to close it here, before invoking `b2`

Great, that makes things simpler.

> I suppose the pythonic way of doing this would be to introduce a scope:
> 
> ```
> with tempfile.NamedTemporaryFile(mode="w+", delete=False) as config:
>    ...
> ```

There's not much benefit to this since nothing is done when the scope exits (because `delete=False`; which we have to override the behavior that the file is deleted when we close it).

However if we don't need to close the file before use, then you're exactly right.